### ISSUE TITLE
Krea 2: explicit INT8 selection can fall back to FP8, making `torch.compile` fail on RTX 30-series

### ISSUE THREAD

**Issue by AZanarddy** [shootthesound/Fizgig#97]
## Summary

When training Krea 2 on an Ampere GPU, selecting **INT8** in the GUI does not always enable the INT8 training path. With `Blocks Swap` manually set to `0`, the GUI can omit `--quant_int8`, so the trainer loads the dynamic FP8 path instead.

If `Compile Blocks` is set to `On`, Inductor/Triton then attempts to compile an FP8 kernel and fails on RTX 30-series GPUs:

```text
ValueError("type fp8e4nv not supported in this architecture.
The supported fp8 dtypes are ('fp8e4b15', 'fp8e5')")
```

This aborts before the first training step.

## Environment

- GPU: NVIDIA RTX 3090, 24 GB, SM 8.6
- Driver: 610.57.04
- PyTorch: 2.10.0+cu128
- CUDA: 12.8
- Fizgig commit: `2fc76fe`

## Reproduction

1. Choose Krea 2 training.
2. Set **Base Precision** to **INT8 — 8-bit, fastest**.
3. Set **Blocks Swap** manually to `0`.
4. Set **Compile Blocks** to `On`.
5. Start a training run.

## Expected behavior

The generated command should include:

```text
--quant_int8 bf16
```

The log should confirm the intended path:

```text
INT8 W8A8 base active
INT8 training path: ... Linears -> W8A8
```

`torch.compile` should compile the INT8 graph without touching an FP8 kernel.

## Actual behavior

The log instead shows:

```text
Loading Krea 2 DiT weights ... (fp8 scaled)
[compile] 28 blocks compiled
```

There is no `INT8 W8A8 base active` message. The first compiled forward then fails with the FP8 Triton architecture error above.

## Likely cause

The GUI only forwards `--quant_int8` through its automatic memory-planning state. Selecting INT8 explicitly while using a manually fixed swap value clears that state, so the explicit precision choice is silently lost and the default FP8 path is used.

At minimum, the command builder should honor the explicit **Base Precision** selection independently of the **Blocks Swap** setting.

## Why this matters / suggested direction

This machine successfully trains the same Krea 2 model at 1024px with compile enabled in AI Toolkit using:

```yaml
qtype: convrot8
compile: true
block_compile: true
```

AI Toolkit's ConvRot8 path is a compile-safe W8A8 implementation: it uses INT8 tensor-core matmul and compile-friendly custom operations, avoiding the FP8 code path entirely. It runs at roughly 18 GB VRAM on this RTX 3090.

Fizgig already has an INT8 training path, but it is not currently selected in this GUI scenario and does not offer the same compile/memory behavior as ConvRot8.

## Proposed acceptance criteria

- Explicit INT8 always emits `--quant_int8 bf16` (or the selected INT8 mode), regardless of `Blocks Swap`.
- Logs clearly state the final resolved precision before model loading.
- On SM 8.6 GPUs, Fizgig never tries to compile the FP8 path.
- Add a regression test for the GUI command builder: explicit INT8 + `blocks_to_swap=0` must produce `--quant_int8`.
- Consider a ConvRot8-style W8A8 backend for Krea 2 training, so compiled INT8 training on Ampere can match the proven AI Toolkit behavior and VRAM use.

**Comment by shootthesound** [comment]
Thanks for this — the diagnosis was spot on. The auto strategy was the only thing that
ever set the INT8 flag, and a manually-set Blocks Swap cleared it, so your explicit pick
was silently dropped exactly as you worked out.

Both halves are fixed on master now, so running `update.bat` gets you going: an explicit INT8
pick emits `--quant_int8 bf16` whenever swap is 0 (however it got there), and the
trainer refuses fp8 + compile outright on SM 8.6 with a plain message instead of the
Triton error. One caveat on "regardless of Blocks Swap": INT8 weights can't ride the
block swap, so INT8 with a manual swap above 0 still runs the fp8 base — but the console
now tells you and how to get INT8 back, rather than changing precision silently. There's
a regression test on the builder for your exact scenario.

The ConvRot8-style compile-safe W8A8 backend is a good shout and it's on the list — some
of the groundwork already exists from other performance work in the codebase.

Closing with the fix — re-open if the updated build doesn't behave on your 3090.

**Comment by shootthesound** [comment]
Follow-up on the ConvRot8-style backend you suggested — benched and built, in two parts.

**Part 1 — the backend already existed.** Krea 2's INT8 base precision is a true W8A8
path: int8 activations x int8 weights through `torch._int_mm` on the int8 tensor cores,
exact bf16 gradients, no fp8 kernels anywhere — architecture-safe on your 3090. What hid
it from you was this very issue: the explicit INT8 pick silently fell to fp8. Measured at
1024px, batch 1: **~18 GB peak — matching the AI Toolkit figure you quoted**, comfortably
inside 24 GB.

**Part 2 — compiled 1024px now fits too, as of today's update.** The blocker was where
the gradient checkpoint sat: compiled *inside* the graph (faster at low res), its stashes
blow past 32 GB at 1 MP. There's now a checkpoint-*outside* boundary — same fused
kernels, eager-level memory. Measured on the same run: **~18.7 GB peak, 2.4 s/step vs
3.3 eager (~27% faster)**, ~25 s compile warm-up. The auto gate places the boundary by
itself, and an explicit Compile=On is now boundary-safe instead of OOMing.

So on your 3090: run `update.bat`, Base Precision INT8, Compile Blocks Auto (or On),
1024px — you should land in ConvRot8's footprint with compiled speed. Would genuinely
love your numbers; this whole thread made the trainer better.

**Comment by AZanarddy** [comment]
This worked beautifully, thank you so much for taking the time to listen, investigate the issue properly, benchmark it, and then build the fix so quickly.

I tested it on my RTX 3090 under CachyOS (Linux Kernel 7.2), which is my daily system. At 1024px, VRAM briefly peaked at 19.3 GB and then settled into a stable 18.7 GB. With Compile Blocks set to On, step time dropped from around 13 s/it to 8.23 s/it. The improvement is genuinely huge.

Training startup improved dramatically too: it used to take about two minutes before training began, and now it is around 50 seconds.

I had wanted to move to Fizgig for quite a while, but this INT8/compile issue was the main blocker for me. It is really great to have it working properly now.

You have been incredibly kind and dedicated throughout this. Thank you again for being so open to the report and for turning it into such a meaningful improvement to the trainer.

**Comment by shootthesound** [comment]
Thanks so much! I'm deligthed its working well for you, please let me know anytime you run into an issue or have an idea for an improvement!

Pete
